DataStructures.Tests/BitListTest.cs

```diff
@@ -0,0 +1,680 @@
+            var array = new bool[1];
+            bits.CopyTo(array, 0);
+            Assert.IsTrue(array.Length == 1 && array[0]);
+        }
+
+        [Test]
+        public static void CopyToByteArrayTest() //void CopyTo(byte[] array, int arrayIndex)
+        {
+            var bits = new BitList((byte)1);
+            var array = new byte[1];
+            bits.CopyTo(array, 0);
+            Assert.IsTrue(array.Length == 1 && array[0] == 1);
+        }
+
+        [Test]
+        public static void CopyToInt16ArrayTest() //void CopyTo(byte[] array, int arrayIndex)
+        {
+            var bits = new BitList((short)1);
+            var array = new short[1];
+            bits.CopyTo(array, 0);
+            Assert.IsTrue(array.Length == 1 && array[0] == 1);
+        }
+
+        [Test]
+        public static void CopyToInt32ArrayTest() //void CopyTo(int[] array, int arrayIndex)
+        {
+            var bits = new BitList(1);
+            var array = new int[1];
+            bits.CopyTo(array, 0);
+            Assert.IsTrue(array.Length == 1 && array[0] == 1);
+        }
+
+        [Test]
+        public static void CopyToInt64ArrayTest() //void CopyTo(long[] array, int arrayIndex)
+        {
+            var bits = new BitList(1L);
+            var array = new long[1];
+            bits.CopyTo(array, 0);
+            Assert.IsTrue(array.Length == 1 && array[0] == 1);
+        }
+
+        [Test]
+        public static void CopySByteArrayTest() //void CopyTo(sbyte[] array, int arrayIndex)
+        {
+            var bits = new BitList((sbyte)1);
+            var array = new sbyte[1];
+            bits.CopyTo(array, 0);
+            Assert.IsTrue(array.Length == 1 && array[0] == 1);
+        }
+
+        [Test]
+        public static void CopyToUInt16ArrayTest() //void CopyTo(ushort[] array, int arrayIndex)
+        {
+            var bits = new BitList((ushort)1);
+            var array = new ushort[1];
+            bits.CopyTo(array, 0);
+            Assert.IsTrue(array.Length == 1 && array[0] == 1);
+        }
+
+        [Test]
+        public static void CopyToUInt32ArrayTest() //void CopyTo(uint[] array, int arrayIndex)
+        {
+            var bits = new BitList(1U);
+            var array = new uint[1];
+            bits.CopyTo(array, 0);
+            Assert.IsTrue(array.Length == 1 && array[0] == 1);
+        }
+
+        [Test]
+        public static void CopyUInt64ArrayTest() //void CopyTo(ulong[] array, int arrayIndex)
+        {
+            var bits = new BitList(1UL);
+            var array = new ulong[1];
+            bits.CopyTo(array, 0);
+            Assert.IsTrue(array.Length == 1 && array[0] == 1);
+        }
+
+        [Test]
+        public static void CopyToSingleArrayTest() //void CopyTo(float[] array, int arrayIndex)
+        {
+            var bits = new BitList(1F);
+            var array = new float[1];
+            bits.CopyTo(array, 0);
+            Assert.IsTrue(array.Length == 1 && array[0] == 1);
```

Comment:
Codacy found an issue: [Do not check floating point equality with exact values, use a range instead.](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=5559704)